messaging/src/main/java/nl/nn/adapterframework/extensions/kafka/KafkaSender.java

```diff
@@ -0,0 +1,99 @@
+   limitations under the License.
+*/
+package nl.nn.adapterframework.extensions.kafka;
+
+import java.util.concurrent.Future;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.ISender;
+import nl.nn.adapterframework.core.PipeLineSession;
+import nl.nn.adapterframework.core.SenderException;
+import nl.nn.adapterframework.core.SenderResult;
+import nl.nn.adapterframework.stream.Message;
+
+/**
+ * Experimental {@link ISender} for sending messages to a Kafka instance.
+ */
+public class KafkaSender extends KafkaFacade implements ISender {
+
+	//setter is for testing purposes only.
+	private @Setter(AccessLevel.PACKAGE) Producer<String, byte[]> producer;
+
+	/** The topic to send messages to. Only one topic per sender. Wildcards are not supported. */
+	private @Setter String topic;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+		if (StringUtils.isEmpty(topic)) throw new ConfigurationException("topic must be specified");
+		if(topic.contains(",")) throw new ConfigurationException("Only one topic is allowed to be used for sender.");
+		if(topic.contains("*")) throw new ConfigurationException("Wildcards are not allowed to be used for sender.");
+	}
+
+	@Override
+	public void open() throws SenderException {
+		producer = new KafkaProducer<>(properties, new StringSerializer(), new ByteArraySerializer());
+		try {
+			Thread.sleep(100);
+		} catch (InterruptedException e) {
+			throw new SenderException(e);
+		}
+		Double metric = (Double) producer.metrics().values().stream().filter(item -> item.metricName().name().equals("response-total")).findFirst().orElseThrow(() -> new SenderException("Failed to get response-total metric.")).metricValue();
+		if (metric.intValue() == 0) throw new SenderException("Didn't get a response from Kafka while connecting for Sending.");
+	}
+
+	@Override
+	public void close() {
+		producer.close();
+	}
+
+	@Override
+	public SenderResult sendMessage(Message message, PipeLineSession session) throws SenderException {
+		byte[] messageData;
+		try {
+			message.preserve();
+			messageData = message.asByteArray();
+		} catch (Exception e) {
+			throw new SenderException("Failed to convert message to message type:", e);
+		}
+		ProducerRecord<String, byte[]> producerRecord = new ProducerRecord<>(topic, messageData);
+		Future<RecordMetadata> future = producer.send(producerRecord);
+		RecordMetadata metadata;
+		try {
+			metadata = future.get();
```

Comment:
We need to be awaiting the response? So sending to Kafka cluster is synchronous.

Reply:
Ok, was just wondering. I'll close it.